zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/ExecutionListenerTest.java

```diff
@@ -756,6 +761,506 @@ public void shouldEvaluateExpressionsForProcessExecutionListeners() {
+
+    // complete subprocess end ELs
+    ENGINE.job().ofInstance(processInstanceKey).withType(END_EL_TYPE + "_sub_1").complete();
+    ENGINE.job().ofInstance(processInstanceKey).withType(END_EL_TYPE + "_sub_2").complete();
+
+    // assert the process instance has completed as expected
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limitToProcessInstanceCompleted())
+        .extracting(
+            r -> r.getValue().getElementId(),
+            r -> r.getValue().getBpmnElementType(),
+            Record::getIntent)
+        .containsSubsequence(
+            tuple(PROCESS_ID, BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_ACTIVATED),
+            tuple(
+                "start_parent",
+                BpmnElementType.START_EVENT,
+                ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple(
+                "task_parent",
+                BpmnElementType.SERVICE_TASK,
+                ProcessInstanceIntent.ELEMENT_ACTIVATED),
+            tuple(
+                "task_parent",
+                BpmnElementType.SERVICE_TASK,
+                ProcessInstanceIntent.ELEMENT_TERMINATED),
+            tuple(
+                subprocessId,
+                BpmnElementType.EVENT_SUB_PROCESS,
+                ProcessInstanceIntent.ELEMENT_ACTIVATING),
+            tuple(
+                subprocessId,
+                BpmnElementType.EVENT_SUB_PROCESS,
+                ProcessInstanceIntent.COMPLETE_EXECUTION_LISTENER),
+            tuple(
+                subprocessId,
+                BpmnElementType.EVENT_SUB_PROCESS,
+                ProcessInstanceIntent.COMPLETE_EXECUTION_LISTENER),
+            tuple(
+                subprocessId,
+                BpmnElementType.EVENT_SUB_PROCESS,
+                ProcessInstanceIntent.ELEMENT_ACTIVATED),
+            tuple(
+                "start_error_sub",
+                BpmnElementType.START_EVENT,
+                ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple("end_sub", BpmnElementType.END_EVENT, ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple(
+                subprocessId,
+                BpmnElementType.EVENT_SUB_PROCESS,
+                ProcessInstanceIntent.ELEMENT_COMPLETING),
+            tuple(
+                subprocessId,
+                BpmnElementType.EVENT_SUB_PROCESS,
+                ProcessInstanceIntent.COMPLETE_EXECUTION_LISTENER),
+            tuple(
+                subprocessId,
+                BpmnElementType.EVENT_SUB_PROCESS,
+                ProcessInstanceIntent.COMPLETE_EXECUTION_LISTENER),
+            tuple(
+                subprocessId,
+                BpmnElementType.EVENT_SUB_PROCESS,
+                ProcessInstanceIntent.ELEMENT_COMPLETED),
+            tuple(PROCESS_ID, BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_COMPLETED));
+  }
```

Comment:
:wrench: I don't see much value in testing multiple types of event subprocesses. The test case and the implementation are the same. 

We could keep one event subprocess test and remove the other two cases.